templates/ignition.yaml

```diff
@@ -38,10 +38,15 @@ systemd:
             Environment="DOCKER_OPT_BIP="
             Environment="DOCKER_OPT_IPMASQ="
             Environment="DOCKER_OPT_MTU="
-{{if (index .TemplatesEnv "http_proxy_enabled")}}
-            Environment=\"HTTP_PROXY=http://{{index .TemplatesEnv "http_proxy.username"}}:{{index .TemplatesEnv "http_proxy.password"}}@{{index .TemplatesEnv "http_proxy.uri"}}:{{index .TemplatesEnv "http_proxy.port"}}\"
-            Environment=\"HTTPS_PROXY=HTTP_PROXY=http://{{index .TemplatesEnv "http_proxy.username"}}:{{index .TemplatesEnv "http_proxy.password"}}@{{index .TemplatesEnv "http_proxy.uri"}}:{{index .TemplatesEnv "http_proxy.port"}}\"
-{{end}}
+            {{- if .TemplatesEnv.http_proxy_enabled }}
+            {{- with .TemplatesEnv.http_proxy }}
+            Environment="HTTP_PROXY=http://{{ .username }}:{{ .password }}@{{ .uri }}:{{ .port }}"
+            Environment="HTTPS_PROXY=http://{{ .username }}:{{ .password }}@{{ .uri }}:{{ .port }}"
+            {{- if .no_proxy }}
+            Environment="NO_PROXY={{ join "," .no_proxy }}"
```

Comment:
Wy do we need this var?

Reply:
`NO_PROXY` is needed if we need to make internal HTTP requests without going through the proxy. This might include talking to the Kubernetes API or something. I can drop it if this isn't something we need to support.

Reply:
no we'll definitely need that - not sure how it will be populated just yet but we can iterate on that